src/tests/java/picard/analysis/directed/CollectTargetedMetricsTest.java

```diff
@@ -0,0 +1,160 @@
+    private final static String library = "TestLibrary1";
+
+    private final static File TEST_DIR = new File("testdata/picard/sam/CollectGcBiasMetrics/");
+    private final File dict = new File(TEST_DIR, "Mheader.dict");
+
+    File tempSamFile;
+
+    SAMRecordSetBuilder setBuilder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    SAMReadGroupRecord readGroupRecord = new SAMReadGroupRecord(readGroupId);
+
+    @Override
+    public String getCommandLineProgramName() {
+        return CollectTargetedPcrMetrics.class.getSimpleName();
+    }
+
+    //create a samfile with 40000 reads for testing whether a cap is found.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final int numReads = 40000;
+        final String readName = "TESTBARCODE";
+
+        tempSamFile = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        final File tempSamFileUnsorted = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        tempSamFileUnsorted.deleteOnExit();
+        tempSamFile.deleteOnExit();
+
+        BufferedLineReader bufferedLineReader = null;
+        try {
+            bufferedLineReader = new BufferedLineReader(new FileInputStream(dict));
+        } catch (final FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        final SAMTextHeaderCodec codec = new SAMTextHeaderCodec();
+        final SAMFileHeader header = codec.decode(bufferedLineReader, dict.toString());
+        header.setSortOrder(SAMFileHeader.SortOrder.unsorted);
+
+        final String separator = ":";
+        final int ID = 1;
+        final int max = 15000;
+        final int min = 1;
+        final Random rg = new Random(5);
+
+        readGroupRecord.setSample(sample);
+        readGroupRecord.setPlatform(platform);
+        readGroupRecord.setLibrary(library);
+        readGroupRecord.setPlatformUnit(readGroupId);
+        header.addReadGroup(readGroupRecord);
+        setBuilder.setReadGroup(readGroupRecord);
+        setBuilder.setUseNmFlag(true);
+
+        setBuilder.setHeader(header);
+
+        for (int i = 0; i < numReads; i++) {
+            final int start = rg.nextInt(max) + min;
+            final String newReadName = readName + separator + ID + separator + i;
+            setBuilder.addPair(newReadName, 0, start+ID, start+ID+99);
+        }
+
+        final SAMFileWriter writer = new SAMFileWriterFactory()
+                .setCreateIndex(true).makeBAMWriter(header, false, tempSamFileUnsorted);
+
+        for (final SAMRecord record : setBuilder) {
+            writer.addAlignment(record);
+        }
+        writer.close();
+
+        //sort the temp file
+        final SortSam sorter = new SortSam();
+        final String[] args = new String[]{"INPUT=" + tempSamFileUnsorted.getAbsolutePath(), "OUTPUT=" + tempSamFile.getAbsolutePath(), "SORT_ORDER=coordinate"};
+
+        sorter.instanceMain(args);
+    }
+
+    @Test //test all collection levels
```

Comment:
Better make each of the 2 functions a @Test and use a @DataProvider providing {tempSamFile} to both of them. (easier to see which test fails if it does)